receiver/datadogreceiver/receiver_test.go

```diff
@@ -171,9 +178,9 @@ func TestDatadogResponse(t *testing.T) {
 			dd.(*datadogReceiver).nextTracesConsumer = consumertest.NewErr(tc.err)
 
 			require.NoError(t, dd.Start(ctx, componenttest.NewNopHost()))
-			t.Cleanup(func() {
+			defer func() {
```

Comment:
Initially I had a comment explaining why a `defer` was being used here, but when removing the redundant `t.Cleanup(cancel)` I noticed that there were already many tests using `defer`, given that I opted to remove the comments on the ones that I added so they just look like the ones in the other tests.